### PR TITLE
refactor suggest_traits_to_import

### PR DESCRIPTION
**Pull request by lcnr**

### CODE REVIEW

File: src/librustc_typeck/check/method/suggest.rs [hunk 1]
```diff
@@ -940,72 +940,69 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
-
-                                        if !fn_sig.decl.implicit_self.has_implicit_self()
-                                            && self_first_arg
-                                        {
-                                            if let Some(ty) = fn_sig.decl.inputs.get(0) {
-                                                arbitrary_rcvr.push(ty.span);
-                                            }
-                                            return false;
-                                        }
+                unsatisfied_predicates.iter().all(|(p, _)| match p {
```
Comment (lcnr) [inline src/librustc_typeck/check/method/suggest.rs:953]: I think I changed this from `~exists x: y` to `forall x: ~y`, but I don't fully understand this check so I might have
messed something up here.

**Comment by lcnr** [comment]
I slightly reduced the scope of the changes. I recommend reviewing this with disabled whitespace diffs.

**Review by davidtwco — APPROVED** [review]
LGTM, but want to get a second opinion

**Pull request merged**